pkg/actions/mustgather.go

```diff
@@ -0,0 +1,364 @@
+		writeContainerLogToFile(containerID, cwContainerName)
+	}
+
+	// Collect Codewind PFE workspace
+	logMG("Collecting Codewind workspace")
+	pfeContainerID := getContainerID(docker.PfeContainerName)
+	copyCodewindWorkspace(pfeContainerID)
+
+	if c.Bool("projects") {
+		// Collect project container inspection & logs
+		dockerClient, dockerErr := docker.NewDockerClient()
+		if dockerErr != nil {
+			HandleDockerError(dockerErr)
+			os.Exit(1)
+		}
+		allContainers, cListErr := docker.GetContainerListWithOptions(dockerClient, types.ContainerListOptions{All: true})
+		if cListErr != nil {
+			HandleDockerError(cListErr)
+			os.Exit(1)
+		}
+		for _, cwContainer := range docker.GetContainersToRemove(allContainers) {
+			logMG("Collecting information from container " + cwContainer.Names[0])
+			writeContainerInspectToFile(cwContainer.ID, filepath.Join("projects", cwContainer.Names[0]))
+			writeContainerLogToFile(cwContainer.ID, filepath.Join("projects", cwContainer.Names[0]))
+		}
+	}
+
+	// Collect docker-compose file
+	logMG("Collecting docker-compose.yaml")
+	copyFileHere(filepath.Join(codewindHome, "docker-compose.yaml"), "docker-compose.yaml")
+
+	// Collect codewind version
+	logMG("Collecting CWCTL version")
+	d1 := []byte(appconstants.VersionNum)
+	versionErr := ioutil.WriteFile(filepath.Join(mustGatherDirName, "cwctl.version"), d1, 0644)
+	if versionErr != nil {
+		errors.CheckErr(versionErr, 201, "")
+	}
+
+	// Attempt to gather Eclipse logs
+	codewindEclipseWSDir := c.String("eclipseWorkspaceDir")
+	if codewindEclipseWSDir != "" {
+		codewindEclipseWSLogDir := filepath.Join(codewindEclipseWSDir, ".metadata")
+		if _, err := os.Stat(codewindEclipseWSLogDir); !os.IsNotExist(err) {
+			files, dirErr := ioutil.ReadDir(codewindEclipseWSLogDir)
+			if dirErr != nil {
+				logMG("Unable to collect Eclipse logs - directory read error " + dirErr.Error())
+			}
+			logMG("Collecting Eclipse Logs")
+			eclipseLogDir := "eclipseLogs"
+			mustGatherEclipseLogPath := filepath.Join(mustGatherDirName, eclipseLogDir)
+			logDirErr := os.MkdirAll(mustGatherEclipseLogPath, 0755)
+			if logDirErr != nil {
+				errors.CheckErr(logDirErr, 205, "")
+			}
+			for _, f := range files {
+				fileName := f.Name()
+				if f.Mode().IsRegular() && strings.HasSuffix(fileName, ".log") {
+					copyFileHere(filepath.Join(codewindEclipseWSLogDir, fileName), filepath.Join(eclipseLogDir, fileName))
+				}
+			}
+		} else {
+			logMG("Unable to collect Eclipse logs - workspace metadata directory not found")
+		}
+	} else {
+		logMG("Unable to collect Eclipse logs - workspace not specified")
+	}
+
+	// Attempt to gather VSCode logs
+	logMG("Collecting VSCode logs")
+	vsCodeDir := ""
+	switch runtime.GOOS {
+	case "darwin":
+		vsCodeDir = filepath.Join(homeDir, "Library", "Application Support", "Code")
+	case "linux":
+		vsCodeDir = filepath.Join(homeDir, ".config", "Code")
+	case "windows":
+		vsCodeDir = filepath.Join(homeDir, "AppData", "Roaming", "Code")
```

Comment:
Does this need a default case for unsupported OSs?

Reply:
No - default case already handled at line 135

Reply:
True, although if the default case put out the log message at line 198 and then returned you could get rid of `if len(vsCodeDir) > 0 {` and outdent everything by one. (This isn't a change request though, it is fine as it is.)